pkg/controller/controller_binding.go

```diff
@@ -349,38 +352,67 @@ func isPlanBindable(serviceClass *v1alpha1.ServiceClass, plan *v1alpha1.ServiceP
 
 func (c *controller) injectBinding(binding *v1alpha1.Binding, credentials map[string]interface{}) error {
 	glog.V(5).Infof("Creating Secret %v/%v", binding.Namespace, binding.Spec.SecretName)
-	secret := &v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      binding.Spec.SecretName,
-			Namespace: binding.Namespace,
-		},
-		Data: make(map[string][]byte),
-	}
 
+	secretData := make(map[string][]byte)
 	for k, v := range credentials {
 		var err error
-		secret.Data[k], err = serialize(v)
+		secretData[k], err = serialize(v)
 		if err != nil {
+			// Terminal error
+			// TODO mark as terminal error once we have the terminal condition
 			return fmt.Errorf("Unable to serialize credential value %q: %v; %s",
 				k, v, err)
 		}
 	}
 
-	found := false
-
-	_, err := c.kubeClient.Core().Secrets(binding.Namespace).Get(binding.Spec.SecretName, metav1.GetOptions{})
-	if err == nil {
-		found = true
+	// Creating/updating the Secret
+	secretClient := c.kubeClient.Core().Secrets(binding.Namespace)
+	found := true
+	existingSecret, err := secretClient.Get(binding.Spec.SecretName, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			// Terminal error
+			// TODO mark as terminal error once we have the terminal condition
+			return err
+		}
+		err = nil
+		found = false
```

Comment:
Minor thing but since `found` is only used in the following if-statement, you might as well just move the "true" part of that if-statement into here. Then you can add an "else" for this `if` that just deals with creating a new one.